scripts/services.js

```diff
@@ -257,7 +257,8 @@ var trackerCaptureServices = angular.module('trackerCaptureServices', ['ngResour
         else{
 
             var startDate = DateUtils.format( moment(referenceDate, calendarSetting.momentFormat).add(offset, 'days') );
-            var periodOffset = _periodOffset && dhis2.validation.isNumber( _periodOffset ) ? _periodOffset : splitDate(startDate).year - splitDate(DateUtils.getToday()).year;
+            var yearsSinceEnrollment = splitDate(DateUtils.getToday()).year - splitDate(startDate).year;
+            var periodOffset = (dhis2.validation.isNumber(_periodOffset) || 0) - yearsSinceEnrollment;

```

Comment:
This line doesn't look quite right? Apparently `periodOffset = _periodOffset` was supposed to be a possible outcome in the original expression, but it is not so here.

Reply:
@superskip To keep the existing logic we could modify this to the following:
`var periodOffset = dhis2.validation.isNumber(_periodOffset) || -yearsSinceEnrollment`

I thought it made more sense to start the suggested periods from the beginning rather than the end, but I agree to keep the original functionality we should go with the above suggestion. If you are happy with that I'll update the PR.

Reply:
@plinnegan  `dhis2.validation.isNumber` returns a boolean, so this line might result in `periodOffset = true`, not `periodOffset = _periodOffset` :open_mouth:

Reply:
@superskip Apologies this should keep the same behaviour:
`var periodOffset = dhis2.validation.isNumber(_periodOffset) ? _periodOffset : -yearsSinceEnrollment`
If `_periodOffset` is `undefined`, this will return `-yearsSinceEnrollment`
If `_periodOffset` is `0` or any other number, this will return that number

Reply:
@plinnegan If the expression doesn't work with undefined inputs, here is a potential fix:
`var periodOffset = dhis2.validation.isNumber(_periodOffset ?? '') ? _periodOffset : -yearsSinceEnrollment`

Reply:
@superskip I don't believe so, because `dhis2.validation.isNumber(_periodOffset)` will return false for `null` or `undefined`. The key change is that the new logic will NOT return the false result from the ternary operator when `_periodOffset=0`, which is what is happening currently.

Reply:
@plinnegan Right, then your last solution looks good to me! :+1:

Reply:
@superskip PR updated, thanks for your patience on this!

Reply:
@plinnegan Likewise, and thanks for fixing it :)